diktat-test-framework/src/main/kotlin/org/cqfn/diktat/test/framework/processing/TestComparatorUnit.kt

```diff
@@ -1,28 +1,39 @@
 package org.cqfn.diktat.test.framework.processing
 
-import org.apache.commons.io.FileUtils
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
-import java.io.File
 import java.io.IOException
-import java.nio.file.Files
+import java.nio.file.Path
 import java.nio.file.Paths
-import java.util.ArrayList
-import java.util.stream.Collectors
+import kotlin.io.path.Path
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.copyTo
+import kotlin.io.path.isRegularFile
+import kotlin.io.path.readLines
 
 /**
  * Class that can apply transformation to an input file and then compare with expected result and output difference.
+ *
+ * @property resourceFilePath only used when the files are loaded as resources,
+ *   via [compareFilesFromResources].
  * @property function a transformation that will be applied to the file
  */
 @Suppress("ForbiddenComment", "TYPE_ALIAS")
 class TestComparatorUnit(private val resourceFilePath: String,
                          private val function: (expectedText: String, testFilePath: String) -> String) {
     /**
-     * @param expectedResult
-     * @param testFileStr
-     * @param trimLastEmptyLine
-     * @return true if transformed file equals expected result, false otherwise
+     * @param expectedResult the name of the resource which has the expected
+     *   content. The trailing newline, if any, **won't be read** as a separate
+     *   empty string. So, if the content to be read from this file is expected
+     *   to be terminated with an empty string (which is the case if
+     *   `newlineAtEnd` is `true`), then the file should end with **two**
+     *   consecutive linebreaks.
+     * @param testFileStr the name of the resource which has the original content.
+     * @param trimLastEmptyLine whether the last (empty) line should be
+     *   discarded when reading the content of [testFileStr].
+     * @return `true` if transformed file equals expected result, `false` otherwise.
+     * @see compareFilesFromFileSystem
```

Comment:
I believe in KDocs preferred way of linking other declarations is via square brackets, then IDEA will also check if this symbol can be resolved

Reply:
Indeed, except for the `@see` tags, where `[]` are unnecessary.